scio-core/src/main/scala/com/spotify/scio/values/PairHashSCollectionFunctions.scala

```diff
@@ -93,16 +93,15 @@ class PairHashSCollectionFunctions[K, V](val self: SCollection[(K, V)])
     leftHashed.map(x => (x._1, x._2)) ++ rightHashed
   }
 
-  private def combineAsMapSideInput[W: ClassTag](that: SCollection[(K, W)])
-  : SideInput[MMap[K, ArrayBuffer[W]]] = {
-    that.combine { case (k, v) =>
+   private lazy val combineAsMapSideInput: SideInput[MMap[K, ArrayBuffer[V]]] = {
```

Comment:
Is this pointless?

Will ```  implicit def makePairHashSCollectionFunctions[K: ClassTag, V: ClassTag](s: SCollection[(K, V)])
  : PairHashSCollectionFunctions[K, V] =
    new PairHashSCollectionFunctions(s)
``` create a new instance every time anyway?

Reply:
Don't think this will actually cache anything. `PairHashSCollectionFunctions` is an implicit class so everytime `that.combineAsMapSideInput` is called the logic below are executed again.

Reply:
I agree with @nevillelyh. It will get re-evaluated every time.